gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -0,0 +1,283 @@
+    proc_var._non_alt_infos = {}
+    annotation_field_set = set(self._annotation_fields or [])
+    for key, variant_info in variant.info.iteritems():
+      # TODO(bashir2): field_count should be removed from VariantInfo and
+      # instead looked up from header_fields.
+      if (self._split_alternate_allele_info_fields and
+              variant_info.field_count == _FIELD_COUNT_ALTERNATE_ALLELE):
+        self._add_per_alt_info(proc_var, key, variant_info.data)
+      elif key in annotation_field_set:
+        self._add_annotation(proc_var, key, variant_info.data)
+      else:
+        proc_var._non_alt_infos[key] = variant_info.data
+    return proc_var
+
+  def _add_per_alt_info(self, proc_var, field_name, variant_info_data):
+    if len(variant_info_data) != len(proc_var._alternate_datas):
+      raise ValueError(
+          'Per alternate INFO field {} does not have same cardinality as '
+          ' number of alternates: {} vs {}'.format(
+              field_name, len(variant_info_data),
+              len(proc_var._alternate_datas)))
+    for alt_index, info in enumerate(variant_info_data):
+      proc_var._alternate_datas[alt_index]._infos[field_name] = info
+
+  def _add_annotation(self, proc_var, field_name, data):
+    # type: (ProcessedVariant, str, List[str]) -> None
+    """Adds an annotation INFO field based on the format in the header.
+
+    Args:
+      proc_var (ProcessedVariant): The object to which the annotations are being
+        added.
+      field_name (str): The name of the annotation field.
+      data(List[str]): The data part of the field separated on comma.
+    """
+    if not self._header_fields.infos.has_key(field_name):
+      raise ValueError('{} INFO not found in the header, variant: {}'.format(
+          field_name, proc_var))
+    header_desc = self._header_fields.infos[field_name].desc
+    annotation_names = extract_annotation_names(header_desc)
+    for annotation_str in data:
+      annotations = _extract_annotation_list_with_alt(annotation_str)
+      if len(annotation_names) != len(annotations) - 1:
+        raise ValueError('Expected {} annotations, got {}'.format(
+            len(annotation_names), len(annotations) - 1))
+      found_alt = False
+      # This assumes that number of alternate bases and annotation segments
+      # are not too big. If this assumption is not true, we should replace the
+      # following loop with a hash table search and avoid the quadratic time.
+      for alt in proc_var._alternate_datas:
+        if alt.alternate_bases == annotations[0]:
+          found_alt = True
+          self._add_annotation_for_alt(
+              alt, field_name, annotations[1:], annotation_names)
+          break
+      # TODO(bashir2): Currently we only check exact matches of alternate bases
+      # which is not enough. We should implement the whole standard for finding
+      # alternate bases for an annotation list.
+      if not found_alt:
```

Comment:
nit: you can use python's "for else" statement for these cases, but this may not be needed anyway if we change how we process the annotations (see above).